sdk/eventhub/event-hubs/test/partitionPump.spec.ts

```diff
@@ -16,4 +30,147 @@ describe("PartitionPump", () => {
     // ...and if they don't override then we give them the default (currently it's EventPosition.earliest())
     getStartingPosition(undefined, undefined).offset!.should.equal(-1);
   });
+
+  describe("telemetry", () => {
+    const eventHubProperties = {
+      endpoint: "theendpoint",
+      eventHubName: "theeventhubname"
+    };
+
+    it("basic span properties are set", async () => {
+      let name: string | undefined;
+      let options: SpanOptions | undefined;
+      const fakeParentSpan = new NoOpSpan();
+      const createdSpan = new SlightlyLessNoOpSpan("myspan");
+
+      const actualReturnedSpan = await createProcessingSpan(
+        [],
+        eventHubProperties,
+        {
+          tracingOptions: {
+            spanOptions: {
+              parent: fakeParentSpan
+            }
+          }
+        },
+        {
+          startSpan: (nameArg, optionsArg) => {
+            name = nameArg;
+            options = optionsArg;
+            return createdSpan;
+          }
+        }
+      );
+
+      should.equal(name, "Azure.EventHubs.process");
+
+      should.exist(options);
+      options!.kind!.should.equal(SpanKind.CONSUMER);
+      options!.parent!.should.equal(fakeParentSpan);
+
+      createdSpan.should.be.equal(actualReturnedSpan);
+      createdSpan.attributes!.should.deep.equal({
```

Comment:
FWIW I record attributes inside of `TestSpan`s now

Reply:
Yes, yes you do. Killed my code. So happy.